risc0/zkvm/src/session.rs

```diff
@@ -92,13 +97,24 @@ pub struct Segment {
     pub insn_cycles: usize,
 }
 
+/// The Events of [Session]
+pub trait Events {
+    /// Fired before the proving of a segment starts.
+    fn on_pre_prove_segment(&self, _: &Segment) {}
+
+    /// Fired after the proving of a segment ends.
+    fn on_post_prove_segment(&self, _: &Segment) {}
+}
```

Comment:
```suggestion
pub trait SessionEvents {
    /// Fired before the proving of a segment starts.
    fn on_pre_prove_segment(&self, segment: &Segment) {}

    /// Fired after the proving of a segment ends.
    fn on_post_prove_segment(&self, segment: &Segment) {}
}
```

Reply:
Thanks for the suggestion. We've renamed `Events` and `_`.